CMakeLists.txt

```diff
@@ -201,6 +201,77 @@ include(cmake/Coverage.cmake)
+    # https://nrn.readthedocs.io/en/latest/scm/guide/SCMGuide.html#a-versioning-scheme-for-neuron.
+    # In a branch whose parent tag is 8.2.dev, the project(...) call should say that the version is
+    # 8.2.0. If and when an incompatible change is merged to the master branch then a 9.0.dev tag
+    # should be created and the project(...) version should be updated to 9.0.0. If changes are
+    # cherry-picked to a branch such as release/8.1 then a 8.1.1.dev tag should be created there and
+    # the project(...) version in that branch updated to 8.1.1. The three cases handled are:
+    #
+    # * if the parent tag is a MAJOR.MINOR.PATCH version then it should exactly match project(...)
+    # * if the parent tag is MAJOR.MINOR(a|b|rcN) then project(...) should contain MAJOR.MINOR.0
+    # * if the parent tag has a .dev suffix, the part before .dev is suffixed with .0 if needed to
+    #   make a full MAJOR.MINOR.PATCH version, which should exactly match project(...)
+    if(PARENT_GIT_TAG MATCHES "^([0-9]+\\.[0-9]+)\\.dev$")
+      # X.Y.dev CMAKE_MATCH_1 holds X.Y
+      set(expected_version "${CMAKE_MATCH_1}.0")
+    elseif(PARENT_GIT_TAG MATCHES "^([0-9]+\\.[0-9]+\\.[0-9]+)\\.dev$")
+      # X.Y.Z.dev, CMAKE_MATCH_1 holds X.Y.Z
+      set(expected_version "${CMAKE_MATCH_1}")
+    elseif(PARENT_GIT_TAG MATCHES "^([0-9]+)\\.([0-9]+)\\.([0-9]+)$")
+      # Release tag
+      set(expected_version "${PARENT_GIT_TAG}")
+    elseif(PARENT_GIT_TAG MATCHES "^([0-9]+\\.[0-9]+)(a|b|rc[0-9]+)$")
+      # Alpha, beta or release candidate (what about alpha/beta/rc of a patch release, i.e. 8.1.1a?
+      # This is not foreseen in the docs.)
+      set(expected_version "${CMAKE_MATCH_1}.0")
+    else()
+      message(FATAL_ERROR "Failed to parse Git tag: '${GIT_STDOUT}'")
+    endif()
+    if(NOT expected_version STREQUAL
+       "${PROJECT_VERSION_MAJOR}.${PROJECT_VERSION_MINOR}.${PROJECT_VERSION_PATCH}")
+      message(WARNING "Parent git tag is ${PARENT_GIT_TAG} but the CMake version is "
+                      "${PROJECT_VERSION}; you should probably update CMakeLists.txt "
+                      "to contain project(NEURON VERSION ${expected_version} ...)!")
+    endif()
+  else()
+    message(
+      STATUS "git describe failed "
+             "(stdout: '${PARENT_GIT_TAG}', stderr: '${GIT_STDERR}', code: ${GIT_EXIT_CODE}), "
+             "skipping the version check")
+  endif()
+endif()
+# Save the version information from project(...) to an nrnsemanticversion.h header in the build
+# directory. This separate from nrnversion.h, which includes information about the latest git
+# commit, because it should change less frequently, but it is included in many more files.
+set(NRN_VERSION_MAJOR ${PROJECT_VERSION_MAJOR})
+set(NRN_VERSION_MINOR ${PROJECT_VERSION_MINOR})
+set(NRN_VERSION_PATCH ${PROJECT_VERSION_PATCH})
+configure_file(src/nrnoc/nrnsemanticversion.h.in src/nrnoc/nrnsemanticversion.h @ONLY)
+
```

Comment:
`nitpick`: could we move this into separate cmake file to keep top level cmakefile bit concise?

Reply:
Done in 2646a64a90f5fbd25b7438344dfd6f5d2e01bd8e